opflexagent/firewall_wrapper.py

```diff
@@ -0,0 +1,181 @@
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import contextlib
+from neutron.agent import firewall as fw
+from oslo_log import log as logging
+from oslo_utils import importutils
+
+
+LOG = logging.getLogger(__name__)
+
+
+class FirewallDriverWrapper(fw.FirewallDriver):
+    """Wrapper for Firewall Driver implemntations
+
+    This wrapper class keeps a map of strategies to
+    use for implementing a firewall driver. The strategy
+    selected is determined by a key, which is passed in
+    a map as part of the init_wrapper function. The strategy
+    must know how to select the appropriate firewall driver
+    based solely on the key associated with that driver. The
+    key should be obtained from the port object, or if no port
+    object is supplied, it should return all of them
+    """
+
+    def __init__(self):
+        self.strategy_map = {}
+        self.strategy_fn = None
+        self.default_strategy = None
+
+    def strategy(self, port):
```

Comment:
is this meant to be a public method? Looks like something you would just use internally